### ISSUE TITLE
js->py type conversions for arrays/lists not using native types

### ISSUE THREAD

**Issue by rth** [iodide-project/pyodide#780]
A round trip of a Python list (for instance of strings) to JS and back produces a JsProxy object instead of the original list,

**Example**
In Python,
```py
x = ['a', 'b']
```

then in JS,
```js
var x = pyodide.pyimport('x')
```
and finally in Python,
```py
import js
type(js.x)  # produces a JsProxy, instead of a list.
```

**Comment by oeway** [comment]
I am a bit confused, do you mean:
```
import js
type(js.x)  # <---------type(js.x) instead of type(js)?
```

**Comment by rth** [comment]
Yes, indeed it was a typo.

**Comment by hoodmane** [comment]
This was resolved by #1167. I just added tests for this in #1188.
